docs/src/explanations/Documentation-of-functions.md

```diff
@@ -0,0 +1,313 @@
+ChiselStage.emitVerilog(new Wrapper)
+```
+### Conclusion:
+When '<>' was used to assign a wire with an unknown direction "tmp" to DontCare, the following error message appeared.
+But when ":=" was used to assign the wire to DontCare, No errors came up. 
+
+
+##  Concept 4: You can use <> or := to assign DontCare to directioned things (IOs)
+':=' or '<>' which is best used to assign directioned things to DontCare? 
+We will find that out using the sample codes above.
+( Scastie link for the ecperiment:https://scastie.scala-lang.org/Shorla/ZIGsWcylRqKJhZCkKWlSIA/1)
+
+```scala mdoc:silent:reset
+import chisel3._
+import chisel3.util.DecoupledIO
+
+class Wrapper extends Module{
+  val io = IO(new Bundle {
+  val in = Flipped(DecoupledIO(UInt(8.W)))
+  val out = DecoupledIO(UInt(8.W))
+  })
+  val p = Module(new PipelineStage)
+  val c = Module(new PipelineStage) 
+  //connect Producer to IO
+  io.in := DontCare
+  p.io.a <> DontCare
+  val tmp = Wire(Flipped(DecoupledIO(UInt(8.W))))
+  tmp <> DontCare
+  p.io.a <> io.in
+  // connect producer to consumer
+  c.io.a <> p.io.b
+  //connect consumer to IO
+  io.out <> c.io.b
+}
+class PipelineStage extends Module{
+  val io = IO(new Bundle{
+    val a = Flipped(DecoupledIO(UInt(8.W)))
+    val b = DecoupledIO(UInt(8.W))
+  })
+  io.b <> io.a
+}
+```
+Below we can see the resulting Verilog for this example:
+```scala modc
+ChiselStage.emitVerilog(new Wrapper)
+```
+### Conclusion: 
+Both "<>" and ":=" can be used to assign directioned things (IOs) to DontCare as shown in "io.in" and "p.io.a" respectively. This is basically equivalent because in this case both "<>" and ":=" will figure the direction from the LHS.
+
+##  Concept 5:  `<>` Cannot be used to connect two non-directioned things (Wires)
+The goal is to check if <> can connect two wires using the Experiment code above.
+( Scastie link for the ecperiment:https://scastie.scala-lang.org/Shorla/ZIGsWcylRqKJhZCkKWlSIA/1)
+
+```scala mdoc:silent:reset
+import chisel3._
+import chisel3.util.DecoupledIO
+
+class Wrapper extends Module{
+  val io = IO(new Bundle {
+  val in = Flipped(DecoupledIO(UInt(8.W)))
+  val out = DecoupledIO(UInt(8.W))
+  })
+  val p = Module(new PipelineStage)
+  val c = Module(new PipelineStage) 
+  //connect Producer to IO
+  io.in := DontCare
+  p.io.a <> DontCare
+  val tmp = Wire(Flipped(DecoupledIO(UInt(8.W))))
+  tmp := DontCare
+  p.io.a <> io.in
+  // connect producer to consumer
+  c.io.a <> p.io.b
+  //connect consumer to IO
+  io.out <> c.io.b
+}
+class PipelineStage extends Module{
+  val io = IO(new Bundle{
+    val a = Flipped(DecoupledIO(UInt(8.W)))
+    val b = DecoupledIO(UInt(8.W))
+  })
+  io.b <> io.a
+}
+```
+Below we can see the resulting Verilog for this example:
+```scala modc:crash
+ChiselStage.emitVerilog(new Wrapper)
+```
+### Conclusion:
+The code above shows that <> can't connect two wires, this is because Chisel can't figure out which way things flow. If it is used this is the expected error.
+
```

Comment:
```suggestion
The code above shows that `<>` can't connect two `Wire`s, this is because Chisel can't figure out which way things flow. If it is used this is the expected error.

```